components/client-attestation-mgt/org.wso2.carbon.identity.client.attestation.mgt/src/main/java/org/wso2/carbon/identity/client/attestation/mgt/services/ClientAttestationServiceImpl.java

```diff
@@ -92,49 +89,40 @@ public ClientAttestationContext validateAttestation(String attestationObject,
                 LOG.debug("App :" + serviceProvider.getApplicationResourceId() + " in tenant : " + tenantDomain +
                         " is not subscribed to Client Attestation Service.");
             }
-            clientAttestationContext.setAttestationEnabled(false);
+            clientAttestationContext = new ClientAttestationContext(false);
             clientAttestationContext.setAttested(true);
-            return clientAttestationContext;
-        }
-
-        // Check if the attestation object is empty.
-        if (StringUtils.isEmpty(attestationObject)) {
-            // App is configured to validate attestation but attestation object is empty.
-            // This is a potential attack, so reject the request.
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("App :" + serviceProvider.getApplicationResourceId() + " in tenant : " + tenantDomain +
-                        " is requested with empty attestation object.");
-            }
-            clientAttestationContext.setAttestationEnabled(true);
-            clientAttestationContext.setAttested(false);
-            clientAttestationContext.setValidationFailureMessage("App is configured to validate attestation " +
-                    "but attestation object is empty.");
-            return clientAttestationContext;
-        }
-
-        if (isAndroidAttestation(attestationObject)) {
-
-            clientAttestationContext.setAttestationEnabled(true);
-            clientAttestationContext.setClientType(Constants.ClientTypes.ANDROID);
-
-            ClientAttestationValidator androidAttestationValidator =
-                    new AndroidAttestationValidator(applicationResourceId, tenantDomain,
-                            serviceProvider.getClientAttestationMetaData());
-            androidAttestationValidator.validateAttestation(attestationObject, clientAttestationContext);
-            return clientAttestationContext;
-        } else if (isAppleAttestation(attestationObject)) {
-            clientAttestationContext.setAttestationEnabled(true);
-            clientAttestationContext.setClientType(Constants.ClientTypes.IOS);
-
-            ClientAttestationValidator appleAttestationValidator =
-                    new AppleAttestationValidator(applicationResourceId, tenantDomain,
-                            serviceProvider.getClientAttestationMetaData());
-            appleAttestationValidator.validateAttestation(attestationObject, clientAttestationContext);
-            return clientAttestationContext;
         } else {
-            handleInvalidAttestationObject(clientAttestationContext);
-            return clientAttestationContext;
+            clientAttestationContext = new ClientAttestationContext(true);
```

Comment:
Let's add
```
        clientAttestationContext.setApplicationResourceId(applicationResourceId);
        clientAttestationContext.setTenantDomain(tenantDomain);
```
here, then it can be used with the flow.